core/src/main/java/org/commonjava/maven/galley/internal/TransferManagerImpl.java

```diff
@@ -943,4 +955,65 @@ public boolean publish( final ConcreteResource resource, final InputStream strea
         return batch;
     }
 
+    private void generateNPMFilesFromTarget( final String content, ConcreteResource resource, final Transfer target,
```

Comment:
We've kept all of the maven-specific pieces out of this library (they're actually in ArtifactManager, in galley-maven). I don't think NPM-specific stuff belongs in galley-core. Is this something we can handle using a TransferDecorator or maybe just in the Indy codebase?

Also, we need to allow the STORAGE_PATH attribute to be used for things that aren't NPM...so the presence of this attribute shouldn't be enough to trigger a whole bunch of assumptions about NPM.

We should be able to intercept the store / create code in the Indy NPM handler, and do something like this...though I'd still prefer to see us write the content to a temp file (NOT in NFS storage if we can avoid it), then read the parts using a stream-oriented parser...and writing the tgz content back to disk without storing it in memory. Again, if all of that is possible.

Tarballs can get pretty big. Keeping these in memory will limit how far Indy can scale when supporting NPM builds.